packages/axelar-soroban-std/src/events.rs

```diff
@@ -0,0 +1,155 @@
+use core::fmt::Debug;
+use soroban_sdk::{Env, IntoVal, Topics, Val};
+
+#[cfg(any(test, feature = "testutils"))]
+pub use testutils::*;
+
+pub trait Event: Debug + PartialEq {
+    fn topics(&self) -> impl Topics + Debug;
+
+    fn data(&self) -> impl IntoVal<Env, Val> + Debug;
+
+    fn emit(&self, env: &Env) {
+        env.events().publish(self.topics(), self.data());
+    }
+}
+
+#[cfg(any(test, feature = "testutils"))]
+mod testutils {
+    use crate::events::Event;
+    use soroban_sdk::testutils::Events;
+    use soroban_sdk::{Env, Val};
+
+    pub trait EventTestutils: Event {
+        fn matches(
+            &self,
+            env: &Env,
+            event: &(soroban_sdk::Address, soroban_sdk::Vec<Val>, Val),
```

Comment:
```suggestion
            event: &(Address, Vec<Val>, Val),
```
don't need to use qualified naming for sdk types